tests/functional/u/undefined/undefined_variable_py38.py

```diff
@@ -180,3 +180,8 @@ def expression_in_ternary_operator_inside_container_wrong_position():
 # Self-referencing
 if (z := z):  # [used-before-assignment]
     z = z + 1
+
+
+if (never_defined := False):
+    pass
+print(never_defined)  # [used-before-assignment]
```

Comment:
This looks like a new false-positive. Assignment expressions are bound `before` the conditional check. Thus this executes just fine and will print `False`.

Reply:
Good catch. Fixed in #7847.